coap-core/src/main/java/com/mbed/coap/server/internal/DuplicationDetector.java

```diff
@@ -129,21 +103,10 @@ public void putResponse(CoapPacket request, CoapPacket response) {
 
     @Override
     public void run() {
-        int removedItems = 0;
-        Iterator<CoapRequestId> it = requestMap.keySet().iterator();
-        final long currentTimeMillis = System.currentTimeMillis();
-        while (it.hasNext()) {
-            if (!it.next().isValid(currentTimeMillis)) {
-                it.remove();
-                removedItems++;
-            }
-        }
-        if (LOGGER.isTraceEnabled() && removedItems > 0) {
-            LOGGER.trace("CoAP request duplicate list, non valid items removed: " + removedItems + " ");
-        }
+        requestMap.clean();
     }
 
-    static class CoapRequestId {
+    static public class CoapRequestId implements ExpiringKey {
```

Comment:
I think I need to remove the validTimeout here? It's redundant if the expiring responsibility is moved to cache.